cfgov/ask_cfpb/models/pages.py

```diff
@@ -241,6 +244,211 @@ def page_js(self):
+        return super(PortalSearchPage, self).get_context(
+            request, *args, **kwargs)
+
+    def get_nav_items(self, request, page):
+        """Return nav items sorted by an arbitrary order."""
+        pk_sort_order = [1, 4, 5, 2, 3]
+        hand_sorted_categories = []
+        for pk in pk_sort_order:
+            category = PortalCategory.objects.get(pk=pk)
+            hand_sorted_categories.append({
+                'title': _(category.heading),
+                'url': "{}{}/".format(page.url, slugify(_(category.heading))),
+                'active': (
+                    False if not page.portal_category
+                    else _(category.heading)
+                    == _(page.portal_category.heading)),
+            })
+        return [{
+            'title': _(page.portal_topic.heading),
+            'url': page.url,
+            'active': False if page.portal_category else True,
+            'expanded': True,
+            'children': hand_sorted_categories
+        }], True
+
+    @route(r'^$')
+    def portal_topic_page(self, request):
+        context = self.get_context(request)
+        heading = self.get_heading(self.portal_topic)
+        context['heading'] = heading
+        search_term = request.GET.get('search_term', '').strip()
+        if not search_term or len(unquote(search_term)) == 1:
+            count = self.query_base.count()
+            sqs = self.query_base
+            search_message = self.results_message(count, heading, search_term)
+        else:
+            sqs = self.query_base.filter(content=search_term)
+            count = sqs.count()
+            if count == 0:
+                # No results, so let's try to suggest a better query
+                suggestion = SearchQuerySet().models(
+                    AnswerPage).spelling_suggestion(search_term)
+                if suggestion == search_term:
+                    suggestion = None
+                elif (request.GET.get('correct', '1') == '1' and
+                        flag_enabled('ASK_SEARCH_TYPOS', request=request)):
+                    sqs = self.query_base.filter(content=suggestion)
+                    search_term, suggestion = suggestion, search_term
+                    count = sqs.count()
+            context['pages'] = sqs
+            search_message = self.results_message(count, heading, search_term)
+        paginator = Paginator(sqs, 10)
+        page_number = validate_page_number(request, paginator)
+        pages = paginator.page(page_number)
+        context.update({
+            'count': count,
+            'search_term': search_term,
+            'results_message': search_message,
+            'pages': pages,
+            'paginator': paginator,
+            'current_page': page_number,
+            'get_secondary_nav_items': self.get_nav_items,
+        })
+        return TemplateResponse(
+            request,
+            self.get_template(request),
+            context)
+
+    @route(r'^(?P<category>[^/]+)/$')
```

Comment:
There's a lot of repeated code here and in `portal_topic_page` above.   Would it be possible to use something reusable like what I did originally in my `see-all` branch: https://github.com/cfpb/cfgov-refresh/compare/see-all#diff-e83ae18f3e2ce1056a644da835042016R289